state/user.go

```diff
@@ -235,13 +254,14 @@ func (st *State) RemoveUser(tag names.UserTag) error {
 				return nil, errors.Trace(err)
 			}
 		}
```

Comment:
You should be building all of those ops and txns above into this buildTxn function so they can be done together.